### ISSUE TITLE
pip install fail building from source v1.1.5 (fixed in master?)

### ISSUE THREAD

**Issue by jjlorenzo** [getsentry/symbolic#20]
## fails
``` sh
root@demr-sentry:/tmp/tmp.oEhGBl # pip install --no-binary :all: symbolic
Collecting symbolic
  Using cached symbolic-1.1.5.zip
Collecting milksnake (from symbolic)
  Downloading milksnake-0.1.1.zip
Collecting cffi>=1.6.0 (from milksnake->symbolic)
  Using cached cffi-1.11.2.tar.gz
Collecting pycparser (from cffi>=1.6.0->milksnake->symbolic)
  Using cached pycparser-2.18.tar.gz
Skipping bdist_wheel for symbolic, due to binaries being disabled for it.
Skipping bdist_wheel for milksnake, due to binaries being disabled for it.
Skipping bdist_wheel for cffi, due to binaries being disabled for it.
Skipping bdist_wheel for pycparser, due to binaries being disabled for it.
Installing collected packages: pycparser, cffi, milksnake, symbolic
  Running setup.py install for pycparser ... done
  Running setup.py install for cffi ... done
  Running setup.py install for milksnake ... done
  Running setup.py install for symbolic ... error
    Complete output from command /usr/bin/python2 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-kez8y9/symbolic/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-B4dWEr-record/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-2.7
    creating build/lib.linux-x86_64-2.7/symbolic
    copying symbolic/proguard.py -> build/lib.linux-x86_64-2.7/symbolic
    copying symbolic/utils.py -> build/lib.linux-x86_64-2.7/symbolic
    copying symbolic/debuginfo.py -> build/lib.linux-x86_64-2.7/symbolic
    copying symbolic/symcache.py -> build/lib.linux-x86_64-2.7/symbolic
    copying symbolic/__init__.py -> build/lib.linux-x86_64-2.7/symbolic
    copying symbolic/sourcemap.py -> build/lib.linux-x86_64-2.7/symbolic
    copying symbolic/exceptions.py -> build/lib.linux-x86_64-2.7/symbolic
    copying symbolic/_compat.py -> build/lib.linux-x86_64-2.7/symbolic
    copying symbolic/minidump.py -> build/lib.linux-x86_64-2.7/symbolic
    copying symbolic/demangle.py -> build/lib.linux-x86_64-2.7/symbolic
    copying symbolic/common.py -> build/lib.linux-x86_64-2.7/symbolic
        Updating registry `https://github.com/rust-lang/crates.io-index`
        Updating git repository `https://github.com/m4b/goblin`
    warning: package replacement is not used: https://github.com/rust-lang/crates.io-index#goblin:0.0.11
       Compiling void v1.0.2
       Compiling pdqsort v0.1.2
       Compiling semver-parser v0.7.0
       Compiling rustc-serialize v0.3.24
       Compiling scroll v0.7.0
       Compiling matches v0.1.6
       Compiling unicode-xid v0.0.4
       Compiling percent-encoding v1.0.1
       Compiling itoa v0.3.4
       Compiling cc v1.0.3
       Compiling serde v1.0.24
       Compiling nodrop v0.1.12
       Compiling utf8-ranges v1.0.0
       Compiling sha1 v0.2.0
       Compiling lazy_static v1.0.0
       Compiling dtoa v0.4.2
       Compiling serde v0.9.15
       Compiling if_chain v0.1.2
       Compiling unicode-normalization v0.1.5
       Compiling byteorder v1.2.1
       Compiling fallible-iterator v0.1.3
       Compiling plain v0.2.3
       Compiling cfg-if v0.1.2
       Compiling rustc-demangle v0.1.5
       Compiling fnv v1.0.6
       Compiling quote v0.3.15
       Compiling num-traits v0.1.41
       Compiling log v0.3.8
       Compiling libc v0.2.34
       Compiling regex-syntax v0.4.1
       Compiling lazy_static v0.2.11
       Compiling gcc v0.3.54
       Compiling stable_deref_trait v1.0.0
       Compiling linked-hash-map v0.4.2
       Compiling symbolic-minidump v1.1.5 (file:///tmp/tmpOEicfA/rustsrc/minidump)
       Compiling unreachable v1.0.0
       Compiling unicode-bidi v0.3.4
       Compiling dmsort v0.1.3
       Compiling synom v0.11.3
       Compiling arrayvec v0.4.6
       Compiling base64 v0.4.2
       Compiling semver v0.6.0
       Compiling num-integer v0.1.35
       Compiling memmap v0.5.2
       Compiling rand v0.3.18
       Compiling memchr v2.0.1
       Compiling owning_ref v0.3.3
       Compiling lru-cache v0.1.1
       Compiling thread_local v0.3.5
       Compiling syn v0.11.11
       Compiling gimli v0.15.0
       Compiling idna v0.1.4
       Compiling num-iter v0.1.34
       Compiling backtrace-sys v0.1.16
       Compiling rustc_version v0.2.1
       Compiling aho-corasick v0.6.4
       Compiling uuid v0.5.1
       Compiling serde_json v0.9.10
       Compiling num-complex v0.1.41
       Compiling symbolic-demangle v1.1.5 (file:///tmp/tmpOEicfA/rustsrc/demangle)
       Compiling num-bigint v0.1.41
       Compiling sourcemap v2.2.0
       Compiling url v1.6.0
       Compiling regex v0.2.3
       Compiling num-rational v0.1.40
       Compiling num v0.1.41
       Compiling backtrace v0.3.4
       Compiling error-chain v0.11.0
       Compiling scroll_derive v0.8.0
       Compiling serde_codegen_internals v0.14.2
       Compiling scroll v0.8.0
       Compiling serde_derive v0.9.15
       Compiling goblin v0.0.13
       Compiling proguard v1.0.0
       Compiling symbolic-common v1.1.5 (file:///tmp/tmpOEicfA/rustsrc/common)
    error[E0277]: the trait bound `errors::Error: std::convert::From<scroll::error::Error>` is not satisfied
       --> /tmp/tmpOEicfA/rustsrc/common/src/errors.rs:126:28
        |
    126 |             Scroll(err) => Error::from(err),
        |                            ^^^^^^^^^^^ the trait `std::convert::From<scroll::error::Error>` is not implemented for `errors::Error`
        |
        = help: the following implementations were found:
                  <errors::Error as std::convert::From<std::io::Error>>
                  <errors::Error as std::convert::From<std::str::Utf8Error>>
                  <errors::Error as std::convert::From<std::num::ParseIntError>>
                  <errors::Error as std::convert::From<errors::ErrorKind>>
                and 7 others
        = note: required by `std::convert::From::from`

    error: aborting due to previous error

    error: Could not compile `symbolic-common`.

    To learn more, run the command again with --verbose.

    ----------------------------------------
Command "/usr/bin/python2 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-kez8y9/symbolic/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-B4dWEr-record/install-record.txt --single-version-externally-managed --compile" failed with error code 101 in /tmp/pip-build-kez8y9/symbolic/
```

## works with master
```
git clone https://github.com/getsentry/symbolic
cd symbolic
git submodule update --init --recursive
cd py
python setup.py bdist_wheel
# create successfully dist/symbolic-1.1.5-py2.py3-none-linux_x86_64.whl
```

``` sh
root@demr-sentry:/ # rustc --version
rustc 1.22.1
root@demr-sentry:/ # cargo --version
cargo 0.22.0
root@demr-sentry:/ # uname -a
Linux demr-sentry 4.9.49-moby #1 SMP Fri Dec 8 13:40:02 UTC 2017 x86_64 Linux
root@demr-sentry:/ # cat /etc/apk/repositories
http://dl-cdn.alpinelinux.org/alpine/edge/main
http://dl-cdn.alpinelinux.org/alpine/edge/community
```

**Comment by jan-auer** [comment]
Hi, thanks for reporting this. It has already been fixed and a release is on the way

**Comment by jan-auer** [comment]
Quick follow-up: We're introducing a couple of breaking changes so the release might actually take a little longer. In the meanwhile, you can always just use the prebuilt wheels that we distribute with every release.

**Comment by mitsuhiko** [comment]
Just a quick heads up: sentry itself does not run on alpine so if the attempt is to make sentry work on alpine that won't go very far :)

**Comment by jjlorenzo** [comment]
Well to be honest, I don't have much knowledge of sentry because I'm new to this wonderful project, but I was able to run sentry on alpine and it seams to be working, I'm missing something?

I had a restricted environment for a client that manage PHI and because the rest of the containers are based on alpine I try installing onpremise using alpine and except `symbolic` compilation, I have not faced big problems so far.

This is my dockerfile
```
FROM alpine:3.7

RUN set -ex &&\
  apk upgrade --no-cache &&\
  echo "@edge-testing http://dl-cdn.alpinelinux.org/alpine/edge/testing" >> /etc/apk/repositories &&\
  echo "@v36-main http://dl-cdn.alpinelinux.org/alpine/v3.6/main" >> /etc/apk/repositories &&\
  apk add --no-cache \
    jansson \
    libcap \
    libjpeg-turbo \
    libpq@v36-main \
    libxslt \
    openjpeg \
    pcre \
    tiff \
    tzdata &&\
  apk add --no-cache \
    libmemcached \
    memcached \
    postgresql-client@v36-main \
    postgresql-contrib@v36-main \
    postgresql@v36-main \
    py2-pip \
    rabbitmq-server@edge-testing \
    redis &&\
  pip install --no-cache-dir --disable-pip-version-check --use-wheel --no-index \
    --find-links=https://private/wheels/sentry/8.22.0 \
    librabbitmq \
    python-memcached \
    sentry \
    sentry-plugins

RUN set -ex &&\
  addgroup -S sentry && adduser -h /var/lib/sentry -G sentry -S -D -s /sbin/nologin -g sentry sentry &&\
  install -d -o nobody -g nobody -m 0755 \
    /var/log/memcached \
    /var/log/postgresql \
    /var/log/rabbitmq \
    /var/log/redis \
    /var/log/sentry-cron \
    /var/log/sentry-web \
    /var/log/sentry-worker &&\
  install -d -o memcached -g memcached -m 0755 /var/lib/memcached

RUN set -ex &&\
  apk add --no-cache \
    curl \
    htop \
    multitail \
    nano

ENV ENV /etc/profile

ENV LANG en_US.utf8

ENV S6_BEHAVIOUR_IF_STAGE2_FAILS 2

ENV S6_LOGGING_SCRIPT "n30 s268435455 T"

ENV TERM xterm-256color

ADD s6-overlay-amd64.tar.gz /

COPY . /

ENTRYPOINT ["/init"]

```

I previously build all sentry wheels using an alpine box prepared for that, here is the dockerfile
```
FROM alpine:3.7

RUN set -ex &&\
  apk update &&\
  apk upgrade &&\
  echo "@edge-testing http://dl-cdn.alpinelinux.org/alpine/edge/testing" >> /etc/apk/repositories &&\
  echo "@v36-main http://dl-cdn.alpinelinux.org/alpine/v3.6/main" >> /etc/apk/repositories &&\
  apk update &&\
  apk add \
    attr-dev \
    build-base \
    cargo \
    curl-dev \
    freetype-dev \
    geoip-dev \
    git \
    jansson-dev \
    jpeg-dev \
    lcms2-dev \
    less \
    libcap-dev \
    libffi-dev \
    libpng-dev \
    libwebp-dev \
    libxml2-dev \
    libxslt-dev \
    linux-headers \
    linux-pam-dev \
    lua5.2-dev \
    ncdu \
    openjpeg-dev \
    paxmark \
    pcre-dev \
    postgresql-dev@v36-main \
    py-setuptools \
    python-dev \
    tiff-dev \
    zeromq-dev \
    zlib-dev &&\
  apk add py2-pip &&\
  pip install --no-cache-dir --disable-pip-version-check wheel

ENV ENV /etc/profile

ENV TERM xterm-256color

COPY rootfs/ /

COPY docker-build docker-run Dockerfile /

WORKDIR dist

```

```
cd `mktemp -d`
git clone https://github.com/getsentry/symbolic
cd symbolic
git submodule update --init --recursive
make wheel
cp py/dist/*.whl /wheels/

cd /dist
rm -r ${OLDPWD}

pip wheel --wheel-dir=`pwd` --find-links=`pwd` sentry
pip wheel --wheel-dir=`pwd` --find-links=`pwd` librabbitmq
pip wheel --wheel-dir=`pwd` --find-links=`pwd` python-memcached

```

**Comment by mitsuhiko** [comment]
Maybe we're in a better state now that we changed some of our dependencies. In the past we were unable to provide an alpine based docker installation because we needed glibc.

**Comment by jjlorenzo** [comment]
Fixed in `2.0.0`, the wheel now correctly compile in alpine. Thanks!

**Comment by vincent-heatseekr** [comment]
This ``sudo apt-get install -y cargo`` did it for me.
